packages/vite/src/node/server/__tests__/pluginContainer.spec.ts

```diff
@@ -106,13 +146,21 @@ async function getPluginContainer(
   inlineConfig?: UserConfig
 ): Promise<PluginContainer> {
   const config = await resolveConfig(
-    { configFile: false, ...inlineConfig },
+    {
+      configFile: false,
+      server: { preTransformRequests: false },
+      ...inlineConfig
+    },
     'serve'
   )
 
   // @ts-ignore: This plugin requires a ViteDevServer instance.
   config.plugins = config.plugins.filter((p) => !/pre-alias/.test(p.name))
 
+  // @ts-ignore: So does this one and this mock one seems to work
+  const iap = config.plugins.find((p) => p.name === 'vite:import-analysis')
+  iap.configureServer(<ViteDevServer>{ moduleGraph })
```

Comment:
Same question here. Can it be moved to another PR?

Reply:
yes, they can, but they will have to be differently, at a broader level perhaps or with heavier and less hacky mock machinery.  here I was trying to stay reasonably close to what I understood to be your idioms in the original tests of https://github.com/vitejs/vite/pull/5465